Projects/TMProject/TMGlobal.h

```diff
@@ -190,7 +190,7 @@ const static TMVector3 g_vecItemBillPos[28] =
     {0.76899999f, 0.509f, -1.143f},
 };
 
-const static stTeleportPos g_TeleportTable[37] =
+const static stTeleportPos g_TeleportTable[42] =
```

Comment:
Vamos aproveitar e mudar para 
```c++
constexpr std::array g_TeleportTable
```

A inicialização fica assim (sem contar com a parte que você adicionou): 
```c++
constexpr std::array g_TeleportTable =
{
  stTeleportPos { 2116, 2100, 700, g_pMessageStringTable[209] },
  stTeleportPos { 2480, 1716, 700, g_pMessageStringTable[209] },
  stTeleportPos { 2456, 2016, 700, g_pMessageStringTable[209] },
  stTeleportPos { 3648, 3108, 0, g_pMessageStringTable[209] },
  stTeleportPos { 1044, 1724, 0, g_pMessageStringTable[125] },
  stTeleportPos { 1044, 1716, 0, g_pMessageStringTable[126] },
  stTeleportPos { 1044, 1708, 0, g_pMessageStringTable[173] },
  stTeleportPos { 1048, 1764, 0, g_pMessageStringTable[210] },
  stTeleportPos { 2140, 2068, 0, g_pMessageStringTable[213] },
  stTeleportPos { 2468, 1716, 0, g_pMessageStringTable[211] },
  stTeleportPos { 2364, 2284, 0, g_pMessageStringTable[212] },
  stTeleportPos { 144, 3788, 0, g_pMessageStringTable[213] },
  stTeleportPos { 2668, 2156, 0, g_pMessageStringTable[212] },
  stTeleportPos { 144, 3772, 0, g_pMessageStringTable[213] },
  stTeleportPos { 148, 3780, 0, g_pMessageStringTable[215] },
  stTeleportPos { 144, 3780, 0, g_pMessageStringTable[215] },
  stTeleportPos { 1004, 4028, 0, g_pMessageStringTable[214] },
  stTeleportPos { 408, 4072, 0, g_pMessageStringTable[215] },
  stTeleportPos { 1004, 4064, 0, g_pMessageStringTable[214] },
  stTeleportPos { 744, 3820, 0, g_pMessageStringTable[215] },
  stTeleportPos { 1004, 3992, 0, g_pMessageStringTable[214] },
  stTeleportPos { 680, 4076, 0, g_pMessageStringTable[216] },
  stTeleportPos { 916, 3820, 0, g_pMessageStringTable[215] },
  stTeleportPos { 876, 3872, 0, g_pMessageStringTable[216] },
  stTeleportPos { 932, 3820, 0, g_pMessageStringTable[215] },
  stTeleportPos { 188, 188, 0, g_pMessageStringTable[126] },
  stTeleportPos { 2548, 1740, 1000, g_pMessageStringTable[322] },
  stTeleportPos { 1824, 1772, 0, g_pMessageStringTable[218] },
  stTeleportPos { 1172, 4080, 0, g_pMessageStringTable[211] },
  stTeleportPos { 1516, 3996, 0, g_pMessageStringTable[220] },
  stTeleportPos { 1304, 3816, 0, g_pMessageStringTable[219] },
  stTeleportPos { 2452, 1716, 0, g_pMessageStringTable[211] },
  stTeleportPos { 2452, 1988, 0, g_pMessageStringTable[211] },
  stTeleportPos { 3648, 3140, 700, g_pMessageStringTable[211] },
  stTeleportPos { 2480, 1648, 700, g_pMessageStringTable[301] },
  stTeleportPos { 1052, 1708, 0, g_pMessageStringTable[321] },
  stTeleportPos { 1056, 1724, 0, g_pMessageStringTable[485] }
};
 ```

Sendo assim, trocaremos os acessos para um range-based for loop ou pelo menos usar o `size()` para iterar usando o índice. Solicito isto porque chequei o código, e você alterou apenas um dos acessos a array `g_TeleportTable`, e alterando já para .len() ou range-based for torna mais difícil de errar no futuro.